discovery.md

```diff
@@ -0,0 +1,448 @@
+
+The CloudEvent `source` attribute is a potential cause of high fanout. For
+example consider a blob storage system where each directory constitutes a
+distinct `source` attribute value. For this reason, `source` is not the starting
+point for discovery, but an attribute that can be used when creating
+subscriptions. A producer is free to show different possible source values
+to different users in their system.
+
+There are use cases where the Discovery API is served as a collection of other
+producers. These take two forms:
+
+1. Aggregator
+2. Events Broker
+
+In the aggregator case, discovery is a combined set of producers for where the
+discovery is passed through largely unmodified and subscriber URIs are generally
+absolute (back to the producer).
+
+In the events broker use case, it is allowable to modify the output of
+discovery and subscription URIs will be relative to the broker itself.
+
+## Notations and Terminology
+
+### Notational Conventions
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD",
+"SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be
+interpreted as described in [RFC 2119](https://tools.ietf.org/html/rfc2119).
+
+### Terminology
+
+This specification defines the following terms:
+
+#### Producer
+
+The "producer" is a specific instance, process or device that creates the data
+structure describing the CloudEvent.
+
+#### Intermediary
+
+An "intermediary" receives a message containing an event for the purpose of
+forwarding it to the next receiver, which might be another intermediary or a
+consumer. A typical task for an intermediary is to route the event to receivers
+based on the information in the event context.
+
+#### Source
+
+The "source" is the context in which the occurrence happened. In a distributed
+system it might consist of multiple Producers. If a source is not aware of
+CloudEvents, an external producer creates the CloudEvent on behalf of the
+source.
+
+#### Consumer
+
+A "consumer" receives the event and acts upon it. It uses the context and data
+to execute some logic, which might lead to the occurrence of new events.
+
+#### Subscription
+
+The request for events from an Event Producer system.
+
+#### Event Subscriber
+
+The entity managing the lifecycle of a Subscription on behalf of an Event
+Consumer. In some instances this might be the same entity as the Event Consumer.
+
+## API Specification
+
+The discovery API takes the declarative API design approach. This document
+structure focuses on discovery where each document has a two part key of
+`producer` and `type`. This takes a denormalized view where the composite key
+of documents is the `producer` and `type`. Producer is defined as an arbitrary
+human readable string and `type` is the CloudEvent `type` attribute.
+
+### Producer entity Attributes
+
+The `Producer` entity is the main component of the discovery API. This section
```

Comment:
Is this producer or source? Referring to the definitions that are listed in the document above.
Producer=The "producer" is a specific instance, process or device that creates the data structure describing the CloudEvent.
Source=The "source" is the context in which the occurrence happened. In a distributed system it might consist of multiple Producers. If a source is not aware of CloudEvents, an external producer creates the CloudEvent on behalf of the source.

Reply:
renamed this throughout to provider.

references to source are gone.